qiskit_nature/drivers/pyscfd/pyscfdriver.py

```diff
@@ -125,49 +302,386 @@ def _check_valid():
+
+            self._mol = gto.Mole(
+                atom=atom,
+                unit=self._unit,
+                basis=self._basis,
+                max_memory=self._max_memory,
+                verbose=verbose,
+                output=output,
             )
-            charge = self.molecule.charge
-            spin = self.molecule.multiplicity - 1
-            units = self.molecule.units.value
+            self._mol.symmetry = False
+            self._mol.charge = self._charge
+            self._mol.spin = self._spin
+            self._mol.build(parse_arg=False)
+
+            if output is not None:
+                self._process_pyscf_log(output)
+                try:
+                    os.remove(output)
+                except Exception:  # pylint: disable=broad-except
+                    pass
+
+        except Exception as exc:
+            raise QiskitNatureError("Failed to build the PySCF Molecule object.") from exc
+
+    @staticmethod
+    def _check_molecule_format(val: str) -> Union[str, List[str]]:
+        """Ensures the molecule coordinates are in XYZ format.
+
+        This utility automatically converts a Z-matrix coordinate format into XYZ coordinates.
+
+        Args:
+            val: the atomic coordinates.
+
+        Raises:
+            QiskitNatureError: If the provided coordinate are badly formatted.
+
+        Returns:
+            The coordinates in XYZ format.
+        """
+        atoms = [x.strip() for x in val.split(";")]
+        if atoms is None or len(atoms) < 1:  # pylint: disable=len-as-condition
+            raise QiskitNatureError("Molecule format error: " + val)
+
+        # An xyz format has 4 parts in each atom, if not then do zmatrix convert
+        # Allows dummy atoms, using symbol 'X' in zmatrix format for coord computation to xyz
+        parts = [x.strip() for x in atoms[0].split(" ")]
+        if len(parts) != 4:
+            try:
+                newval = []
+                for entry in gto.mole.from_zmatrix(val):
+                    if entry[0].upper() != "X":
+                        newval.append(entry)
+                return newval
+            except Exception as exc:
+                raise QiskitNatureError("Failed to convert atom string: " + val) from exc
+
+        return val
+
+    def run_pyscf(self) -> None:
+        """Runs the PySCF calculation.
+
+        This method is part of the public interface to allow the user to easily overwrite it in a
+        subclass to further tailor the behavior to some specific use case.
+
+        Raises:
+            QiskitNatureError: If an invalid HF method type was supplied.
+        """
+        try:
+            hf_method_name = self._hf_method.upper()
+            hf_method_cls = getattr(scf, hf_method_name)
+        except AttributeError as exc:
+            raise QiskitNatureError("Failed to load {} HF object.".format(hf_method_name)) from exc
+
+        self._calc = hf_method_cls(self._mol)
+
+        if "KS" in hf_method_name:
```

Comment:
Since we have enum here we should just be able to ask if the enum is one of the set of them that should be acted upon - that way we don't have some string "KS" as comparison that is we changed enum names, with set the refactoring would tale care of things.

Reply:
The reason why I went with this comparison style is that I did not to duplicate half of the Enum as a set here. And this comparison properly reflects the nature of the check which we care about.

Reply:
I guess in my mind it makes the code 'brittle' in that refactoring an enum, should we ever want to change any name, this detection here would break. Listing the set and simply saying `if x in [A, B, C]` works and if we refactor C to D that code should get changed by the refactor whereas some check on "C" would not.